File: Dante2/Module2.02/Zadanie2.01/main.c
#include <stdlib.h>
#include <stdio.h>
#include <math.h>
#include <time.h>

struct point_t
{
    int x;
    int y;
};

struct point_t* set(struct point_t* p, int x, int y);
void show(const struct point_t* p);
struct point_t* read(struct point_t* p);
float distance(const struct point_t* p1, const struct point_t* p2, int *err_code);

int main()
{
    srand(time(NULL));

    struct point_t p1;
    struct point_t p2;

    int x = 1 + rand() % 10;
    int y = 1 + rand() % 10;

    if(set(&p2, x, y) == NULL)
    {
        printf("Incorrect input");
        return 1;
    }

    if(read(&p1) == NULL)
    {
        printf("Incorrect input");
        return 1;
    }

    show(&p1);
    show(&p2);

    int err_code;
    float result = distance(&p1, &p2, &err_code);
    if(result == -1.0f)
    {
        printf("Incorrect input");
        return 1;
    }

    printf("%.2f",result);

    return 0;
}

float distance(const struct point_t* p1, const struct point_t* p2, int *err_code)
{
    if (p1 == NULL || p2 == NULL)
    {
        if (err_code == NULL)
        {
            return -1.0f;
        }
        *err_code = 1;
        return -1.0f;
    }

    float distance = sqrtf((powf((float)p2->x - (float)p1->x, 2)) + (powf((float)p2->y - (float)p1->y, 2)));
    if (err_code == NULL)
    {
        return distance;
    }

    *err_code = 0;
    return distance;
}

void show(const struct point_t* p)
{
    if(p == NULL)
    {
        return;
    }

    printf("x: %i y: %i\n",p->x, p->y);
}

struct point_t* set(struct point_t* p, int x, int y)
{
    if(p == NULL)
    {
        return NULL;
    }

    p->x = x;
    p->y = y;

    return p;
}

struct point_t* read(struct point_t* p)
{
    if(p == NULL)
    {
        return NULL;
    }

    int check;
    int x, y;

    printf("Enter: ");
    check = scanf("%i", &x);
    if(!check)
    {
        return NULL;
    }

    printf("Enter: ");
    check = scanf("%i",&y);
    if(!check)
    {
        return NULL;
    }

    p->x = x;
    p->y = y;

    return p;
}

File: Dante2/Module2.02/Zadanie2.01/main.rs
use rand::random_range;

struct PointT {
    x: i32,
    y: i32,
}

fn show(p: & PointT) {
    println!("x: {} and y: {}", p.x, p.y);
}

fn distance(p1: & PointT, p2: & PointT) -> f32 {
    let distance: f32 = f32::sqrt((
        f32::powi(p2.x as f32 - p1.x as f32, 2)) +
        f32::powi(p2.x as f32 - p1.x as f32, 2));

    distance
}

fn set(p: &mut PointT, x: i32, y: i32) -> &mut PointT {
    p.x = x;
    p.y = y;

    p
}

fn read(p: &mut PointT) -> &mut PointT {
    let mut input: String = String::new();

    println!("Enter: ");
    std::io::stdin().read_line(&mut input).expect("Incorrect input");
    let x: i32 = input.trim().parse().expect("Incorrect parsing");

    input.clear();

    println!("Enter: ");
    std::io::stdin().read_line(&mut input).expect("Incorrect input");
    let y: i32 = input.trim().parse().expect("Incorrect parsing");

    p.x = x;
    p.y = y;

    p
}

fn main() {
    let num_x: i32 = random_range(0..10);
    let num_y: i32 = random_range(0..10);

    let mut p1 = PointT {
        x: 0,
        y: 0,
    };

    let mut p2 = PointT {
        x: 0,
        y: 0,
    };

    set(&mut p2, num_x, num_y);
    read(&mut p1);

    show(&mut p1);
    show(&mut p2);

    let result: f32 = distance(&p1, &p2);
    println!("Result is: {}", result);
}
